### ISSUE TITLE
OAuth2 with Flow("accessCode") on UI 3.x

### ISSUE THREAD

**Issue by heldersepu** [swagger-api/swagger-ui#3624]
| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  Bug (Maybe)
| Which Swagger/OpenAPI version?  | 2
| Which Swagger-UI version?       | 3.x
| How did you install Swagger-UI? | Swagger-Net & Swashbuckle
| Which browser & version?        | Chrome
| Which operating system?         | Win7

### Demonstration UI 
Here is the 3.x UI sample that throws the error
http://swagger-netoauth2.azurewebsites.net/swagger
And here is the same under 2.x
http://swashbuckleoauth2.azurewebsites.net/swagger

### Expected Behavior
There should not be any errors when we click on Authorize button.

### Current Behavior
Getting errors:
```diff
- Auth Warning Authorization may be unsafe, passed state was changed in server Passed state wasn't returned from auth server
- Auth Error Authorization failed: no accessCode received from the server
```

### Context
I've been simplifying the supported auth (Basic, ApiKey & OAuth2) configurations and ran into an issue with OAuth2. It seems that it was working before under 2x but now on 3x we get errors.
...Or it could be an error on my migration to 3.x

**Comment by heldersepu** [comment]
@shockey @webron Guys I think really could use your help troubleshooting this. 
Any pointers will be greatly appreciated.

I was looking with Fiddler and noticed calls to the oauth2-redirect when if I open that url I get an error in the console:
**Uncaught TypeError: Cannot read property 'swaggerUIRedirectOauth2' of null**
http://swagger-netoauth2.azurewebsites.net/swagger/ui/oauth2-redirect-html?code=AQB04bQyyeM8LByKDXqBPBNVNLAfxVOaRIK_cYHhvHPqwPSxFhsk51H2mQP72YZvRZjkZOOCANT0-DCG2Kf5ayIInOX3Z8yVuBNNiorUwTiNHzkjROSGzl-8VYhtodlA24_fAejt629xnU-cDSDNx-BFd_4dTwncG4jzGpbX_y8MKN8x0n4hJhEK1Wz_5gq_TIOng5q_rayS54esNJP9P3S_sB7ifHwsKuVj_A539-TsP_NaJ2I1830JOxSKuN0-ZrSW-EFjo4FeTlfd6uN_5bqK84JrjPsnoT-sntCVxd0ynokeWOd0G_9Eep7r0LlJiep9Nig4KYnsXKc7-lIYxPnZ_jj4wC-v6f00b2A5VDe0Vg&state=U2F0IFNlcCAwMiAyMDE3IDE0OjU0OjAxIEdNVC0wNDAwIChFYXN0ZXJuIERheWxpZ2h0IFRpbWUp

**Comment by heldersepu** [comment]
Also noticed that the state on the older older version is always ```oauth2``` 

- http://swashbuckleoauth2.azurewebsites.net/swagger/ui/o2c-html?code=AQDT2ITo2byQ0...ecGxA&state=oauth2
- http://swashbuckleoauth2.azurewebsites.net/swagger/ui/o2c-html?code=AQBNRdMZZpO...X4cijuA&state=oauth2

The new one seem to always be something new

- http://swagger-netoauth2.azurewebsites.net/swagger/ui/oauth2-redirect-html?code=AQAI9WLJes...R2gvg&state=U2F0IFNlcCAwMiAyMDE3IDE0OjUyOjMxIEdNVC0wNDAwIChFYXN0ZXJuIERheWxpZ2h0IFRpbWUp
- http://swagger-netoauth2.azurewebsites.net/swagger/ui/oauth2-redirect-html?code=AQB04bQyye...VDe0Vg&state=U2F0IFNlcCAwMiAyMDE3IDE0OjU0OjAxIEdNVC0wNDAwIChFYXN0ZXJuIERheWxpZ2h0IFRpbWUp

Here is my Fiddler session
[oauth2.zip](https://github.com/swagger-api/swagger-ui/files/1272480/oauth2.zip)

**Comment by heldersepu** [comment]
### Found the problem:
- Before we had: [2.x/dist/o2c.html#L3](https://github.com/swagger-api/swagger-ui/blob/2.x/dist/o2c.html#L3)
```js
var qp = null;
if(/code|token|error/.test(window.location.hash)) {
  qp = location.hash.substring(1);
}
else {
  qp = location.search.substring(1);
}
```

- Now we have a simpler version: [v3.1.7/dist/oauth2-redirect.html#L14](https://github.com/swagger-api/swagger-ui/blob/v3.1.7/dist/oauth2-redirect.html#L14)
```js
qp = (window.location.hash || location.search).substring(1);
```

I did test removing the ```window.location.hash || ``` and bye-bye errors.
Yes it is a BUG no maybe anymore...

**Comment by jmsvl** [comment]
Helder helped me pinpoint the source of this bug and I'm also waiting for a fix. Is there any idea of an estimated time of repair?
Thanks

**Comment by shockey** [comment]
This was fixed in #3632, the fix should be released this Friday 🎉 

Closing!
